### PR TITLE
disable "download results" and "edit question" from xray dashcards that have not been saved

### LINKED ISSUE #33637: Can't download results or edit question from an (unsaved) x-ray dashboard
### Describe the bug

Cards on x-ray (unsaved) dashboards have a 3 dots menu with the options of downloading results and of editing the question, but both options fail

image

### To Reproduce

1. Go to "Browse Data"
2. Select the sample database
3. Select "X-ray this table" on the Accounts table
4. Try to download the results or edit any dashcard, for example the one for Total accounts

Edit -> it will prompt to start from scratch, instead of using the selected question

Download results -> `<h1>Bad Message 400</h1><pre>reason: Ambiguous URI empty segment</pre>`

https://github.com/metabase/metabase/assets/1914270/89f710e1-2a19-45c6-bf38-2c4e1e7276ef

### Expected behavior

_No response_

### Logs

_No response_

### Information about your Metabase installation

```JSON
{
  "browser-info": {
    "language": "en-GB",
    "platform": "MacIntel",
    "userAgent": "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/116.0.0.0 Safari/537.36",
    "vendor": "Google Inc."
  },
  "system-info": {
    "file.encoding": "UTF-8",
    "java.runtime.name": "OpenJDK Runtime Environment",
    "java.runtime.version": "20.0.1",
    "java.vendor": "Homebrew",
    "java.vendor.url": "https://github.com/Homebrew/homebrew-core/issues",
    "java.version": "20.0.1",
    "java.vm.name": "OpenJDK 64-Bit Server VM",
    "java.vm.version": "20.0.1",
    "os.name": "Mac OS X",
    "os.version": "13.4",
    "user.language": "en",
    "user.timezone": "Europe/Rome"
  },
  "metabase-info": {
    "databases": [
      "h2",
      "postgres"
    ],
    "hosting-env": "unknown",
    "application-database": "h2",
    "application-database-details": {
      "database": {
        "name": "H2",
        "version": "2.1.214 (2022-06-13)"
      },
      "jdbc-driver": {
        "name": "H2 JDBC Driver",
        "version": "2.1.214 (2022-06-13)"
      }
    },
    "run-mode": "dev",
    "version": {
      "date": "2023-08-29",
      "src_hash": "6f83c596625ba3280d3c8d5407151ecff1f4f8ba",
      "tag": "v0.47.1-SNAPSHOT",
      "branch": "whats-new-ui",
      "hash": "c37e175"
    },
    "settings": {
      "report-timezone": null
    }
  }
}
```

### Severity

-

### Additional context

Discussion on slack: https://metaboat.slack.com/archives/C01LQQ2UW03/p1693397200503659

A quick solution could be to not show the menu for now, as all menu options are not working

### PR DESCRIPTION
**Pull request by npretto**
### Description

closes https://github.com/metabase/metabase/issues/33637

This disables the dashcard menu for unsaved xrays dashboards as suggested here: https://metaboat.slack.com/archives/C01LQQ2UW03/p1693474228115549?thread_ts=1693397200.503659&cid=C01LQQ2UW03

### CODE REVIEW

**Review by ranquild — COMMENTED** [review]
@npretto I'd probably follow the same approach we use for "public" and "embed" dashboards where a boolean flag is added. Seems more robust. I.e. I'd add a new flag here
https://github.com/metabase/metabase/blob/7820266e7d551b038df60b5e8317b16b7ed7a90e/frontend/src/metabase/dashboard/containers/AutomaticDashboardApp.jsx#L168

similar to https://github.com/metabase/metabase/blob/31e83ed96dbe8d26b81a62db25ee841b5f6b163b/frontend/src/metabase/public/containers/PublicDashboard.jsx#L161

If you feel confident we could refactor this flag to a enum value `public/embed/xray` in another PR

File: frontend/src/metabase/dashboard/components/DashCard/DashCardVisualization.tsx [hunk 1]
```diff
@@ -193,7 +193,10 @@ function DashCardVisualization({
         !isEditing &&
         DashCardMenu.shouldRender({ question, result: mainSeries }));
 
-    if (isInternalQuery || !shouldShowDownloadWidget) {
+    const isInXray =
```
[the remark this answers is not in the document]
Comment (npretto) [reply frontend/src/metabase/dashboard/components/DashCard/DashCardVisualization.tsx]: @ranquild thanks I passed the value down as a prop from AutomaticDahsboardApp.
Regarding putting it inside the shouldRender function: if I understand it correctly that function is not even use when the dashboard is in embedded via JWT, should we worry about it?
I can't understand why we have some checks inside shouldRender and some outside, should we put all of them inside shouldRender?

File: frontend/src/metabase/dashboard/components/DashCard/Dashcard.unit.spec.tsx [hunk 2]
```diff
@@ -78,6 +78,12 @@ describe("DashCard", () => {
     expect(screen.getByText("My Card")).toBeVisible();
   });
 
+  it("should not display the ellipsis menu for (unsaved) xray dashboards (metabase#33637)", async () => {
```
Comment (npretto) [inline frontend/src/metabase/dashboard/components/DashCard/Dashcard.unit.spec.tsx:81]: I put the unit test here as it's the component closest to the menu I found that was already setup with mocks etc

File: e2e/test/scenarios/dashboard/reproductions/33637-xray-should-not-have-download-results-edit.cy.spec.js [hunk 3]
```diff
@@ -0,0 +1,33 @@
+import { main, navigationSidebar, restore } from "e2e/support/helpers";
```
Comment (npretto) [inline e2e/test/scenarios/dashboard/reproductions/33637-xray-should-not-have-download-results-edit.cy.spec.js:1]: I will remove this test before merging, for now I'll keep to double check if i break something when changing things

File: frontend/src/metabase/dashboard/components/DashCard/DashCardVisualization.tsx [hunk 4]
```diff
@@ -186,14 +187,16 @@ function DashCardVisualization({
     const question = new Question(dashcard.card, metadata);
     const mainSeries = series[0] as unknown as Dataset;
 
-    const isInternalQuery = question.query() instanceof InternalQuery;
-    const shouldShowDownloadWidget =
-      isEmbed ||
-      (!isPublic &&
-        !isEditing &&
-        DashCardMenu.shouldRender({ question, result: mainSeries }));
+    const shouldShowDownloadWidget = DashCardMenu.shouldRender({
```
Comment (kamilmielnik) [inline frontend/src/metabase/dashboard/components/DashCard/DashCardVisualization.tsx:190]:
```suggestion
    const shouldShowDashCardMenu = DashCardMenu.shouldRender({
```

**Review by kamilmielnik — APPROVED** [review]
LGTM :+1:

**Pull request merged**